utcp.c

```diff
@@ -1042,6 +1053,7 @@ ssize_t utcp_recv(struct utcp *utcp, const void *data, size_t len) {
 
 	// 3. Check incoming data for acceptable seqno and update connection timer
 
+	size_t datalen = len;
```

Comment:
What's the reason for copying `len` here? It's a bit confusing that it changes names and makes me think `len` and `datalen` are now used for different things, but I don't see where.